### ISSUE TITLE
v2.20 crashes loading child collection

### ISSUE THREAD

**Issue by LarryB12** [jcachat/EntityFramework.DynamicFilters#71]
Hi

After upgrading from 1.4.10.2 to v2.20

When running the following code;

MyData.Entry(entity).Collection("ChildCmContacts").Load();

where entity is a CmContact entity and ChildCmContacts is an ICollection<CmContact>
I get a NullReferenceException

A global filter is declared in the OnModelCreating procedure of the DBContext
modelBuilder.Filter("DeletedFilter", (IBaseObject d) =>d.Deleted, false);

This code works in 1.4.10.2

Listed Below ; Callstack, exception and entity model;

**CallStack Attached**
![callstack](https://cloud.githubusercontent.com/assets/20558098/16986924/3517ef48-4e81-11e6-8262-c4dfd087fa70.png)

**Exception**
System.NullReferenceException was unhandled by user code
  HResult=-2147467261
  Message=Object reference not set to an instance of an object.
  Source=EntityFramework.DynamicFilters
  StackTrace:
       at EntityFramework.DynamicFilters.DynamicFilterQueryVisitorCSpace.FindFiltersForEntitySet(ReadOnlyMetadataCollection`1 metadataProperties, EntityContainer entityContainer)
       at EntityFramework.DynamicFilters.DynamicFilterQueryVisitorCSpace.Visit(DbScanExpression expression)
       at System.Data.Entity.Core.Common.CommandTrees.DbScanExpression.Accept[TResultType](DbExpressionVisitor`1 visitor)
       at System.Data.Entity.Core.Common.CommandTrees.DefaultExpressionVisitor.VisitExpression(DbExpression expression)
       at System.Data.Entity.Core.Common.CommandTrees.DefaultExpressionVisitor.VisitExpressionBinding(DbExpressionBinding binding)
       at System.Data.Entity.Core.Common.CommandTrees.DefaultExpressionVisitor.VisitExpressionBindingEnterScope(DbExpressionBinding binding)
       at System.Data.Entity.Core.Common.CommandTrees.DefaultExpressionVisitor.Visit(DbFilterExpression expression)
       at System.Data.Entity.Core.Common.CommandTrees.DbFilterExpression.Accept[TResultType](DbExpressionVisitor`1 visitor)
       at System.Data.Entity.Core.Common.CommandTrees.DefaultExpressionVisitor.VisitExpression(DbExpression expression)
       at System.Data.Entity.Core.Common.CommandTrees.DefaultExpressionVisitor.VisitExpressionBinding(DbExpressionBinding binding)
       at System.Data.Entity.Core.Common.CommandTrees.DefaultExpressionVisitor.VisitExpressionBindingEnterScope(DbExpressionBinding binding)
       at System.Data.Entity.Core.Common.CommandTrees.DefaultExpressionVisitor.Visit(DbProjectExpression expression)
       at System.Data.Entity.Core.Common.CommandTrees.DbProjectExpression.Accept[TResultType](DbExpressionVisitor`1 visitor)
       at System.Data.Entity.Core.Common.CommandTrees.DefaultExpressionVisitor.VisitExpression(DbExpression expression)
       at System.Data.Entity.Core.Common.CommandTrees.DefaultExpressionVisitor.VisitExpressionBinding(DbExpressionBinding binding)
       at System.Data.Entity.Core.Common.CommandTrees.DefaultExpressionVisitor.Visit(DbJoinExpression expression)
       at System.Data.Entity.Core.Common.CommandTrees.DbJoinExpression.Accept[TResultType](DbExpressionVisitor`1 visitor)
       at System.Data.Entity.Core.Common.CommandTrees.DefaultExpressionVisitor.VisitExpression(DbExpression expression)
       at System.Data.Entity.Core.Common.CommandTrees.DefaultExpressionVisitor.VisitExpressionBinding(DbExpressionBinding binding)
       at System.Data.Entity.Core.Common.CommandTrees.DefaultExpressionVisitor.VisitExpressionBindingEnterScope(DbExpressionBinding binding)
       at System.Data.Entity.Core.Common.CommandTrees.DefaultExpressionVisitor.Visit(DbProjectExpression expression)
       at System.Data.Entity.Core.Common.CommandTrees.DbProjectExpression.Accept[TResultType](DbExpressionVisitor`1 visitor)
       at EntityFramework.DynamicFilters.DynamicFilterInterceptor.TreeCreated(DbCommandTreeInterceptionContext interceptionContext)
       at System.Data.Entity.Infrastructure.Interception.DbCommandTreeDispatcher.<Created>b__0(IDbCommandTreeInterceptor i, DbCommandTreeInterceptionContext c)
       at System.Data.Entity.Infrastructure.Interception.InternalDispatcher`1.Dispatch[TInterceptionContext,TResult](TResult result, TInterceptionContext interceptionContext, Action`2 intercept)
       at System.Data.Entity.Infrastructure.Interception.DbCommandTreeDispatcher.Created(DbCommandTree commandTree, DbInterceptionContext interceptionContext)
       at System.Data.Entity.Core.Common.DbProviderServices.CreateCommandDefinition(DbCommandTree commandTree, DbInterceptionContext interceptionContext)
       at System.Data.Entity.Core.Objects.Internal.ObjectQueryExecutionPlanFactory.CreateCommandDefinition(ObjectContext context, DbQueryCommandTree tree)
       at System.Data.Entity.Core.Objects.Internal.ObjectQueryExecutionPlanFactory.Prepare(ObjectContext context, DbQueryCommandTree tree, Type elementType, MergeOption mergeOption, Boolean streaming, Span span, IEnumerable`1 compiledQueryParameters, AliasGenerator aliasGenerator)
       at System.Data.Entity.Core.Objects.EntitySqlQueryState.GetExecutionPlan(Nullable`1 forMergeOption)
       at System.Data.Entity.Core.Objects.ObjectQuery`1.<>c__DisplayClass7.<GetResults>b__6()
       at System.Data.Entity.Core.Objects.ObjectContext.ExecuteInTransaction[T](Func`1 func, IDbExecutionStrategy executionStrategy, Boolean startLocalTransaction, Boolean releaseConnectionOnSuccess)
       at System.Data.Entity.Core.Objects.ObjectQuery`1.<>c__DisplayClass7.<GetResults>b__5()
       at System.Data.Entity.SqlServer.DefaultSqlExecutionStrategy.Execute[TResult](Func`1 operation)
       at System.Data.Entity.Core.Objects.ObjectQuery`1.GetResults(Nullable`1 forMergeOption)
       at System.Data.Entity.Core.Objects.ObjectQuery`1.Execute(MergeOption mergeOption)
       at System.Data.Entity.Core.Objects.DataClasses.EntityCollection`1.Load(List`1 collection, MergeOption mergeOption)
       at System.Data.Entity.Core.Objects.DataClasses.EntityCollection`1.Load(MergeOption mergeOption)
       at System.Data.Entity.Core.Objects.DataClasses.RelatedEnd.Load()
       at System.Data.Entity.Internal.InternalNavigationEntry.Load()
       at System.Data.Entity.Infrastructure.DbCollectionEntry.Load()
       at FwBaseMVVM.BaseViewModels.DataContextAwareViewModelBase.OnLoadEntityChildren(Object entity) in C:\Projects\Source\CH.Framework.02\FwBaseMVVM\FwBaseMVVM\BaseViewModels\DataContextAwareViewModelBase.cs:line 55
       at FmContactManager.ViewModels.VmContactManager.LoadContactGraph(CmContact contact) in C:\Projects\Source\CH.Framework.02\Modules\FmContactManager\ViewModels\VmContactManager.cs:line 1693
       at FmContactManager.ViewModels.VmContactManager.set_MainContact(CmContact value) in C:\Projects\Source\CH.Framework.02\Modules\FmContactManager\ViewModels\VmContactManager.cs:line 145
       at FmContactManager.ViewModels.VmContactManager..ctor(IUnityContainer container, IParentViewModelBase parentViewModel) in C:\Projects\Source\CH.Framework.02\Modules\FmContactManager\ViewModels\VmContactManager.cs:line 574
       at FmContactManager.FmContactManagerService.ReturnVmContactManager(IParentViewModelBase registerViewModel) in C:\Projects\Source\CH.Framework.02\Modules\FmContactManager\FmContactManagerService.cs:line 18
       at SBPFClientRegister.ViewModels.VmClientEdit.PostConstruction() in C:\Projects\Source\iProg.SimonBeaujiePictureFrames\SBPFClientRegister\ViewModels\VmClientEdit.cs:line 91
       at SBPFClientRegister.ViewModels.VmClientEdit..ctor(IUnityContainer container, Int32 id) in C:\Projects\Source\iProg.SimonBeaujiePictureFrames\SBPFClientRegister\ViewModels\VmClientEdit.cs:line 72
       at SBPFClientRegister.Decorations.ClientOrderProcessingHeaderDecoratorService.GetViewModel(Int32 id) in C:\Projects\Source\iProg.SimonBeaujiePictureFrames\SBPFClientRegister\Decorations\ClientOrderProcessingHeaderDecoratorService.cs:line 25
       at FmOrderProcessing.ViewModels.VmOpDocumentManager.ManualAddDecorators() in C:\Projects\Source\CH.Framework.02\Modules\FmOrderProcessing\ViewModels\VmOpDocumentManager.cs:line 227
       at FwBaseMVVM.ViewModelBase.ProcessDecorators() in C:\Projects\Source\CH.Framework.02\FwBaseMVVM\FwBaseMVVM\BaseClasses\ViewModelBase.cs:line 216
       at FwBaseMVVM.ViewModelBase.PostConstruction() in C:\Projects\Source\CH.Framework.02\FwBaseMVVM\FwBaseMVVM\BaseClasses\ViewModelBase.cs:line 193
       at FwBaseMVVM.BaseViewModels.ProcessingViewModelBase`1.PostConstruction() in C:\Projects\Source\CH.Framework.02\FwBaseMVVM\FwBaseMVVM\BaseViewModels\ProcessingViewModelBase.cs:line 73
       at FmOrderProcessing.ViewModels.VmOpDocumentManager.PostConstruction() in C:\Projects\Source\CH.Framework.02\Modules\FmOrderProcessing\ViewModels\VmOpDocumentManager.cs:line 172
       at FmOrderProcessing.ViewModels.VmOpDocumentManager..ctor(IUnityContainer container, Int32 id) in C:\Projects\Source\CH.Framework.02\Modules\FmOrderProcessing\ViewModels\VmOpDocumentManager.cs:line 167
  InnerException: 

**The Entity**
using System.Collections.Generic;
using System.ComponentModel;
using System.ComponentModel.DataAnnotations.Schema;
using FwBase.Attributes;
using FwBaseEntityFramework;
using FwInterfaces.ContactManager;

namespace FmContactManager
{
    public class CmContact : BusinessObject, ICmContact
    {
# region ReadOnly Properties

```
    [NotMapped]
    [Browsable(false)]
    public string Name
    {
        get
        {
            if (CmContactType == null)
                return "New Record";

            switch (CmContactType.ContactTypeEnum)
            {
                case ContactType.Person:
                    if (CmContactPersonDetail != null)
                        return CmContactPersonDetail.FullName;
                    return "Null Person Detail!";
                case ContactType.Organisation:
                    if (CmContactOrganisationDetail != null)
                        return CmContactOrganisationDetail.Name;
                    return "Null Organisation Detail";
                default:
                    return "Undefined Type";
            }
        }
    }

    [DisplayName("Type And Name")]
    [NotMapped]
    public string TypeAndName
    {
        get
        {
            if (CmContactType != null)
                return CmContactType.Description + " : " + Name;
            return "NULL Contact Type";
        }
    }

    [DisplayName("Contact Type")]
    [NotMapped]
    [Browsable(false)]
    public string ContactTypeDescription
    {
        get
        {
            if (CmContactType != null)
                return CmContactType.Description;
            return "NULL Contact Type";
        }
    }

    private System.Guid _tempLookup = System.Guid.NewGuid();

    [Browsable(false)]
    [NotMapped]
    public System.Guid TempLookup
    {
        get { return _tempLookup; }
        set { _tempLookup = value; }
    }
```
# endregion \ReadOnly Properties

```
    public CmContact()
    {
        ChildCmContacts = new HashSet<CmContact>();
        CmCommunicationMethods = new HashSet<CmCommunicationMethod>();
        CmAddresses = new HashSet<CmAddress>();
    }

    #region Database Fields

    #endregion Database Fields

    #region Associations

    [Browsable(false)]
    public int? CmContactTypeID { get; set; }

    private CmContactType cmContactType;
    //[Required]
    [Browsable(false)]
    public virtual CmContactType CmContactType
    {
        get { return cmContactType; }
        set { CheckPropertyChanged(ref cmContactType, value); }
    }


    [Browsable(false)]
    public virtual int? CmContactOrganisationDetailID { get; set; }

    private CmContactOrganisationDetail _cmContactOrganisationDetail;

    [Browsable(false)]
    public virtual CmContactOrganisationDetail CmContactOrganisationDetail
    {
        get { return _cmContactOrganisationDetail; }
        set { CheckPropertyChanged(ref _cmContactOrganisationDetail, value); }
    }


    [Browsable(false)]
    public virtual int? CmContactPersonDetailID { get; set; }

    private CmContactPersonDetail _cmContactPersonDetail;

    [Browsable(false)]
    public virtual CmContactPersonDetail CmContactPersonDetail
    {
        get { return _cmContactPersonDetail; }
        set { CheckPropertyChanged(ref _cmContactPersonDetail, value); }
    }

    [Browsable(false)]
    [EntityChildCollection]
    public virtual ICollection<CmContact> ChildCmContacts { get; set; }

    [Browsable(false)]
    [EntityChildCollection]
    public virtual ICollection<CmCommunicationMethod> CmCommunicationMethods { get; set; }

    [Browsable(false)]
    [EntityChildCollection]
    public virtual ICollection<CmAddress> CmAddresses { get; set; }

    #endregion //Associations

    #region Custom Error Checking

    public override void RunCustomErrorDetection()
    {
        if (CmContactType == null)
            AddError("CmContactType", "Please select a contact type");
        if (CmContactOrganisationDetail != null)
            if (!CmContactOrganisationDetail.Deleted)
                CmContactOrganisationDetail.ValidateNow();
        if (CmContactPersonDetail != null)
            if (!CmContactPersonDetail.Deleted)
                CmContactPersonDetail.ValidateNow();

        foreach (var cmCommunicationMethod in CmCommunicationMethods)
        {
            cmCommunicationMethod.ValidateNow();
        }
        foreach (var cmAddress in CmAddresses)
        {
            cmAddress.ValidateNow();
        }
        foreach (var childCmContact in ChildCmContacts)
        {
            childCmContact.ValidateNow();
        }
        base.RunCustomErrorDetection();
    }

    public override string GetChildErrors()
    {
        string error = "";
        if (CmContactOrganisationDetail != null)
            if (CmContactOrganisationDetail.HasErrors)
                error = error + CmContactOrganisationDetail.Error;
        if (CmContactPersonDetail != null)
            if (CmContactPersonDetail.HasErrors)
                error = error + CmContactPersonDetail.Error;
        foreach (var cmCommunicationMethod in CmCommunicationMethods)
        {
            if (cmCommunicationMethod.HasErrors)
                error = error + cmCommunicationMethod.Error;
        }
        foreach (var cmAddress in CmAddresses)
        {
            if (cmAddress.HasErrors)
                error = error + cmAddress.Error;
        }
        foreach (var childCmContact in ChildCmContacts)
        {
            error = error + childCmContact.Error;
        }
        return error;
    }

    #endregion //Custom Error Checking
}
```

}

**Comment by LarryB12** [comment]
Having done a bit more checking, normal child collections load, but this particular child collection is a collection of entities from the same table as the parent.

CmContact
--> ChildCmContacts
            CmContact
            CmContact

**Comment by jcachat** [comment]
I'm still not able to reproduce this.  Even with a child collection of the same type as the parent.

And I don't see anything in the method named in the stack trace (DynamicFilterQueryVisitorCSpace.FindFiltersForEntitySet) that should be able to cause a null exception unless something has changed the entity metadata on us.

2 suggestions:
1) Condense this down to the bare minimum to reproduce and send/post that.  I obviously can't reconstruct your entire environment as it is and there is something else going on that is causing the problem that I'm not seeing.
2) Pull down the source code and build against that instead of the NuGet package.  When you hit the error in the debugger, see if you can figure out exactly why the null exception is happening in this situation.

**Comment by LarryB12** [comment]
Hi!

I have downloaded the source but can't build due to the following error

Error	1	Cannot import the following key file: DynamicFiltersSigningKey.pfx. The key file may be password protected. To correct this, try to import the certificate again or manually install the certificate to the Strong Name CSP with the following key container name: VS_KEY_2BCDCF545CEED72B	EntityFramework.DynamicFilters

I'm not sure how to fix this

**Comment by jcachat** [comment]
Open the properties for the DynamicFilters project and turn off code signing.

**Comment by LarryB12** [comment]
The bit that is crashing is...

var annotations = configuration.GetType().GetProperty("Annotations").GetValue(configuration, null) as Dictionary<string, object>;

from;

        private IEnumerable<DynamicFilterDefinition> FindFiltersForEntitySet(ReadOnlyMetadataCollection<MetadataProperty> metadataProperties, EntityContainer entityContainer)
        {
            var configuration = metadataProperties.FirstOrDefault(p => p.Name == "Configuration")?.Value;
            if (configuration == null)
                return new List<DynamicFilterDefinition>();

            _**var annotations = configuration.GetType().GetProperty("Annotations").GetValue(configuration, null) as Dictionary<string, object>;**_
            if (annotations == null)
                return new List<DynamicFilterDefinition>();

            var filterList = annotations.Select(a => a.Value as DynamicFilterDefinition).Where(a => a != null).ToList();

            //  Note: Prior to the switch to use CSpace (which was done to allow filters on navigation properties),
            //  we had to remove filters that exist in base EntitySets to this entity to fix issues with 
            //  Table-per-Type inheritance (issue #32).  In CSpace none of that is necessary since we are working
            //  with the actual c# models now (in CSpace) so we always have the correct filters and access to all
            //  the inherited properties that we need.

            return filterList;
        }

**Comment by LarryB12** [comment]
If it helps you can remote to my PC and have a look

**Comment by jcachat** [comment]
Compare the data in the properties that are being accessed with a case where it works correctly.  Especially a case for the same type of object in that child collection.  Try to figure out why something is null that is not supposed to be.

Nothing on that line should be null unless something is changing the entity metadata outside of my control.

You don't specify which part of that line is null but I'm going to guess it's the GetProperty("Annotations").  If that is null, then something is removing the annotations data on the entity - which is where the DynamicFilterDefinitions are stored.  Do you have any Conventions configured into your modelBuilder that may be doing that (maybe to handle something special about this ChildCmContacts collection)?

Or see if you can create a small test project that you can send to me that reproduces the problem.  Without being able to reproduce it, all I can do is guess.  I'm not able to remote right now.

**Comment by LarryB12** [comment]
Aha!
I removed 

            modelBuilder.Entity<CmContact>().HasMany(x => x.ChildCmContacts);

from the ModelBuilder and it all works. 
It did work with v1.4.11

Lance

**Comment by LarryB12** [comment]
I will keep you updated if I find anything else.

Many thanks for your help.

Lance

**Comment by jcachat** [comment]
That's not good that doing that would remove our filter config from the entity...  But it's not something I can do anything about either.

Were you setting up your filters before or after that FK?  If before, try moving it after.  In my project, I define all filters after I have finished doing everything else on the model.

**Comment by LarryB12** [comment]
Unfortunately Applying the Filter first or last made no difference

**Comment by LarryB12** [comment]
Hi please find attached an example I worked on;

You probably know this but for anyone else following;

update App.config with your sql connection

    <defaultConnectionFactory type="System.Data.Entity.Infrastructure.SqlConnectionFactory, EntityFramework" >
      <parameters>
        <parameter value="Data Source=larry-beast\sqlserver2008r2;Database=SBPFWPFMetaData;Trusted_Connection=true;" />
      </parameters>
    </defaultConnectionFactory>

create the database context by typing into the Package Manager Console

update-database -verbose

(the dbcontext creates seed data)

to check the effect of the filter, comment out modelBuilder.Entity<Contact>().HasMany(x => x.ChildContacts);

        protected override void OnModelCreating(DbModelBuilder modelBuilder)
        {
            base.OnModelCreating(modelBuilder);

            modelBuilder.Entity<Contact>().HasMany(x => x.ChildContacts);

            modelBuilder.Filter("DeletedFilter", (IBaseObject d) => d.Deleted, false);
        }


Run the Application

[EntityFrameworkDynamicFilters.zip](https://github.com/jcachat/EntityFramework.DynamicFilters/files/377857/EntityFrameworkDynamicFilters.zip)

**Comment by LarryB12** [comment]
I modified the FindFiltersForEntitySet to;

        private IEnumerable<DynamicFilterDefinition> FindFiltersForEntitySet(ReadOnlyMetadataCollection<MetadataProperty> metadataProperties, EntityContainer entityContainer)
        {
            var configuration = metadataProperties.FirstOrDefault(p => p.Name == "Configuration")?.Value;
            if (configuration == null)
                return new List<DynamicFilterDefinition>();


            var d = configuration.GetType().GetProperty("Annotations");
            if(d==null)
                return new List<DynamicFilterDefinition>();

            var annotations = d.GetValue(configuration, null) as Dictionary<string, object>;
            if (annotations == null)
                return new List<DynamicFilterDefinition>();

            var filterList = annotations.Select(a => a.Value as DynamicFilterDefinition).Where(a => a != null).ToList();

            //  Note: Prior to the switch to use CSpace (which was done to allow filters on navigation properties),
            //  we had to remove filters that exist in base EntitySets to this entity to fix issues with 
            //  Table-per-Type inheritance (issue #32).  In CSpace none of that is necessary since we are working
            //  with the actual c# models now (in CSpace) so we always have the correct filters and access to all
            //  the inherited properties that we need.

            return filterList;
        }

which on the surface fixes the exception.

**Comment by jcachat** [comment]
Thanks - that helps a lot.  I thought about making that same type of change to avoid the exception but I'm worried that may just silently skip a filter.  I will look it over the weekend and let you know what I find.

**Comment by LarryB12** [comment]
When debugging, the FindFiltersForEntitySet is called five times (once for each entity) but when the filter HasMany directive is active the FindFiltersForEntitySet function crashes on about its 4-5 call.. but not the first second or third time...

**Comment by akordeon** [comment]
I'm getting an error when trying to access child collections with a many to many relationship.

Things work fine when including the child collection in the query.

	var topic = db.Topics
			.Include(t => t.Labels)
			.Single(t => t.Id == 1);

But without the include, I receive an error.

	var topic = db.Topics.Single(t => t.Id == 1);
	var entry = db.Entry(topic);
	entry.Collection(t => t.Labels).Load();

[error.pdf](https://github.com/jcachat/EntityFramework.DynamicFilters/files/389106/error.pdf)

Here are my DbContext and models.

    using EntityFramework.DynamicFilters;
    using System.Collections.Generic;
    using System.ComponentModel.DataAnnotations;
    using System.Data.Entity;

    namespace FilterTest
    {
        public class FilterTestContext : DbContext
        {
            public DbSet<Topic> Topics { get; set; }
            public DbSet<Label> Labels { get; set; }

            protected override void OnModelCreating(DbModelBuilder modelBuilder)
            {
                modelBuilder.Entity<Topic>()
                    .HasMany(t => t.Labels)
                    .WithMany(l => l.Topics)
                    .Map(
                        m =>
                        {
                            m.MapLeftKey("TopicId");
                            m.MapRightKey("LabelId");
                            m.ToTable("Topic_Label");
                        });

                modelBuilder.Filter("Deleted", (ISoftDelete d) => d.Deleted, false);
            }
        }

        public class Topic : ISoftDelete
        {
            public Topic()
            {
                Labels = new HashSet<Label>();
            }

            [Key]
            public int Id { get; set; }

            [Required]
            public string Title { get; set; }

            public bool Deleted { get; set; }

            public virtual ICollection<Label> Labels { get; set; }
        }

        public class Label
        {
            public Label()
            {
                Topics = new HashSet<Topic>();
            }

            [Key]
            public int Id { get; set; }

            [Required]
            public string Description { get; set; }

            public virtual ICollection<Topic> Topics { get; set; }
        }

        public interface ISoftDelete
        {
            bool Deleted { get; set; }
        }
    }

**Comment by jcachat** [comment]
Sorry for the delay, been swamped.

I finally had a chance to check this out some more.  I found that the reason that line is throwing a null exception is that when doing an entry.Load() on the child collection, a different type of configuration object is being passed in for the generated many-to-many table.  That object does not have an "Annotations" property.  So it is safe to silently skip past it.  If there are filters that need to be applied to the other end of the relationship, they are still handled by another call.

I just pushed out a fix.  Give this a try and let me know if you still have problems.
